libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts

```diff
@@ -1,46 +1,69 @@
 import { BaseValue } from '../../persistence/domain/base-classes/base-value';
 import { Type } from '../../../domain/utility';
+import { Composite } from './base-classes';
 
-export function instanceForType<T>(Type: Type<T>, fromState = {}) {
+// TODO: Add Constraints <T extends object, K extends object = {}>
+export function instanceForType<T, K>(Type: Type<T>, fromState: K): T {
     const newState = new Type(fromState);
 
-    return new Proxy(newState, new StateHandler(fromState));
+    return new Proxy((newState as unknown) as object, new StateHandler()) as any;
 }
 /**
- *  Proxy wrapper
+ *  Proxy wrapper for Entity/Value Object
  */
-class StateHandler<TModel> {
-    constructor(
-        /** Plain entity dto object */
-        private state: TModel
-    ) {
-    }
-
-    get(target: any, prop: PropertyKey): any {
-        // Check is nested object and wraps a new one proxy
-        if (this.state[prop] && typeof this.state[prop] === 'object') {
-            return new Proxy(this.state[prop], this);
+class StateHandler<T extends object, K extends {}> implements ProxyHandler<T> {
+    get(target: T, propertyKey: string | symbol, receiver?: any): any {
+        // Composite.value takes precedence
+        if (target instanceof Composite) {
+            const value = target.value;
+            // if accessed property refers to the object create new wrapper
+            if (propertyKey in value && typeof value[propertyKey] === 'object') {
+                return new Proxy(value[propertyKey], new StateHandler());
+            }
+            // Make sure composite.value contains needed property
+            // otherwise redirect to the default value.
+            // Need it to make working entity.getter flow
+            if (propertyKey in value) {
+                return Reflect.get(value, propertyKey, value);
+            }
         }
-
-        return Reflect.get(target as any, prop);
+        // default flow
+        return Reflect.get(target, propertyKey, receiver);
```

Comment:
Did not we discuss that we will not use Reflect.***

Reply:
Yes. But are you using we need it ? With you previous asnwer ?

Reply:
sorry didn't get your last question